substrate/frame/assets/src/lib.rs

```diff
@@ -456,6 +464,33 @@ pub mod pallet {
 		ValueQuery,
 	>;
 
+	#[pallet::storage]
+	/// Merklized distribution of an asset.
+	pub(super) type CountForMerklizedDistribution<T: Config<I>, I: 'static = ()> =
+		StorageValue<_, u32, ValueQuery>;
+
+	#[pallet::storage]
+	/// Merklized distribution of an asset.
+	pub(super) type MerklizedDistribution<T: Config<I>, I: 'static = ()> = StorageMap<
```

Comment:
nit: could be a `CountedStorageMap` to count active distributions.

Reply:
no, because the count would go down when we clean up a distribution. and we want the count to remain unique